File: Spoj/main/ANARC09A.cpp
using namespace std;
#include<iostream>
#include<cstdio>
#include<cstdlib>
#include<stack>
int main()
{       
	long Case=0;
	while(1){
		Case++;
		char inp[2005],v1,v2;
		stack<char> sta;
		int operations=0;
		scanf("%s",inp);
		if(inp[0]=='-')break;
		for(int i=0;inp[i]!='\0';i++){
			if(sta.empty()){
				sta.push(inp[i]);
			}
			else if(inp[i]=='}'&&sta.top()=='{')		
				sta.pop();
			else 
				sta.push(inp[i]);
		}
		while(!sta.empty()){
			v1=sta.top();sta.pop();
			v2=sta.top();sta.pop();
			if(v1==v2) operations+=1;
			else operations+=2;
		}
		printf("%ld. %d\n",Case,operations);
	}
	return 0;
}

File: Spoj/main/ANARC09A.py
Case=0
while 1:
    Case+=1
    s=raw_input()
    if s[0]=='-':break
    stack=[]
    top=operations=0
    stack.append(s[0])
    for i in range(1,len(s)):
        if top<0:
            stack.append(i)
            top+=1
            continue
        if s[i]=='}':
            if stack[top]=='{':
                stack.pop()
                top-=1
            else:
                stack.append(s[i])
                top+=1
        else:
            stack.append(s[i])
            top+=1
    #print stack
    while(len(stack)>0):
        v1=stack.pop()
        v2=stack.pop()
        if v1==v2:operations+=1
        else:operations+=2
    print '%d. %d' % (Case,operations)
    
        
